opnode/rollup/derive/payload_attributes.go

```diff
@@ -180,25 +181,31 @@ func UserDeposits(height uint64, receipts []*types.Receipt) ([]*types.DepositTx,
 	return out, nil
 }
 
-func BatchesFromEVMTransactions(config *rollup.Config, txs []*types.Transaction) (out []BatchData) {
+func BatchesFromEVMTransactions(config *rollup.Config, txs []*types.Transaction) (out []*BatchData) {
 	l1Signer := config.L1Signer()
 	for _, tx := range txs {
 		if to := tx.To(); to != nil && *to == config.BatchInboxAddress {
 			seqDataSubmitter, err := l1Signer.Sender(tx)
 			if err != nil {
+				// TODO: log error
 				continue // bad signature, ignore
 			}
 			// some random L1 user might have sent a transaction to our batch inbox, ignore them
 			if seqDataSubmitter != config.BatchSenderAddress {
 				continue // not an authorized batch submitter, ignore
 			}
-			out = append(out, ParseBatches(tx.Data())...)
+			batches, err := DecodeBatches(config, bytes.NewReader(tx.Data()))
+			if err != nil {
+				// TODO: log error
```

Comment:
Let's do this before merging?

Reply:
I say we return the error here and handle logging later. This is library code anyway, which IMO shouldn't log.